test/test_SensorFilter.py

```diff
@@ -25,6 +25,13 @@ def test_invalid_nonetype(self):
         expected_type = float
         self.assertFalse(SensorFilter.filter(reading, lowerbound, upperbound, expected_type))
 
+    def test_bad_type(self):
+        reading = 3
+        lowerbound = 1.0
+        upperbound = 5.0
+        expected_type = float
+        self.assertFalse(SensorFilter.filter(reading, lowerbound, upperbound, expected_type))
```

Comment:
If the reading is an int do we want to not update the field? Would anything break if the reading was an int instead of a float? Maybe you could just cast it to a float.

Reply:
I checked that there is nothing stopping a field from being the wrong type, for example, min_voltage is `float32` (https://github.com/UBCSailbot/sailbot-msg/blob/master/msg/min_voltage.msg) yet I can do this:

```console
$ python
Python 2.7.17 (default, Mar 18 2022, 13:21:42) 
[GCC 7.5.0] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from sailbot_msg.msg import min_voltage
>>> m = min_voltage(min_voltage=3)
>>> type(m)
<class 'sailbot_msg.msg._min_voltage.min_voltage'>
>>> type(m.min_voltage)
<type 'int'>

```

Reply:
This is a valuable corner case that I did not consider. I will look into resolving this.